Source/CombatExtended/Harmony/Harmony_JobGiver_OptimizeApparel.cs

```diff
@@ -0,0 +1,34 @@
+﻿using HarmonyLib;
+using RimWorld;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Verse;
+using Verse.AI;
+
+namespace CombatExtended.HarmonyCE
+{
+    /* Intercepts the outfit manager's attempts to equip a shield or shield belt.
+     * If the current loadout has a projectile weapon in it, does not let the outfit manager grab a shield belt.
+     * If the current loadout has a shield in it, does not let the outfit manager grab a shield.
+     */
+    [HarmonyPatch(typeof(JobGiver_OptimizeApparel), "ApparelScoreGain_NewTmp")]
+    internal static class Harmony_JobGiver_OptimizeApparel_ApparelScoreGain_NewTmp
+    {
+        internal static bool Prefix(Pawn pawn, Apparel ap, ref float __result)
```

Comment:
Shouldn't this patch also check if the pawn is equipped with a two-handed weapon, and return -1000f if the item is Apparel_Shield? Right now colonists with a two-handed weapon will still go grab a shield if their outfit allows it, requiring manual intervention by the player.